contents/data/sql/SQL.md

```diff
@@ -0,0 +1,243 @@
+Here are some of the main reasons behind the widespread adoption of SQL and why it still remains popular today.
+
+### Easy to learn
+
+SQL is very easy to learn, even for beginners who do not have any prior experience with databases. Since it has been around for a few decades, there are many good books and online resources to learn from.
+In addition, SQL and its distributions have a huge support community (such as [Stack Overflow](https://stackoverflow.com/questions/tagged/sql) and the [official MySQL forum](https://forums.mysql.com/)) which can prove useful when you run into problems while using SQL.
+
+### Free of cost
+
+One of the benefits (and reasons) for SQL's popularity is that there are free distributions available (like MySQL, PostgreSQL and SQLite) as well as paid ones (like Microsoft SQL Server and Oracle) that come with more functionality.
+
+One such distribution is MySQL. Since it is an open source software, you do not have to pay to use MySQL. It even comes with official ([MySQL Workbench](https://dev.mysql.com/doc/workbench/en/)) as well third party easy-to-use <tooltip content="Graphical User Interface">GUIs</tooltip>, which are less daunting to new users as compared to a <tooltip content="Command Line Interface">CLI</tooltip>.
+
+### Compatibility
+
+SQL distributions work on many operating systems and more importantly, can be integrated with many languages. This includes languages like PHP, PERL, C++, Ruby, Java as well as *relatively "newer"* ones like Python, JavaScript (Node.js) and Go.
+
+In addition, online playgrounds like [DB Fiddle](https://www.db-fiddle.com/) and [JDoodle](https://www.jdoodle.com/execute-sql-online) make it easy to use and learn the SQL language quickly without the hassle of setting up an environment or application.
+
+## Disadvantages
+
+SQL is not without its problems. In general, the biggest problem is with regards to the features of SQL.
+
+Although SQL databases use established <tooltip content="American National Standard Institutes">ANSI</tooltip> & <tooltip content="International Organization for Standardization">ISO</tooltip> standards, some distributions add proprietary extensions to the standard SQL to ensure customer lock-in.
+Thus, the available feature set varies according to the SQL distribution that you are using. This can make it confusing and frustrating to use SQL when switching across distributions.
+
+Apart from that, most **SQL** problems are not uniformly applicable to all of its distributions.
+
+For example, **MySQL** suffers from concurrency issues. Though it performs well with read operations, it *can* be problematic when there are many concurrent read-write operations. A symptom of this issue would be a sudden slowdown of a well-optimized query. However, **PostgreSQL** deals with concurrency well by being fully <tooltip content="Atomicity, Consistency, Isolation, Durability properties are satisfied">ACID</tooltip> compliant and implementing transactions isolation.
```

Comment:
Apart from concurrency, are there any other disadvantages of using Sql as a developer

Reply:
I feel that this is covered in the preceding paragraph